### PR TITLE
Scaffold application from database

### PR DESCRIPTION
**Pull request by efirs**
`tigris scaffold go/ts/java {db_name}`

Outputs application code with schemas generated for all collections of
the specified database.

### CODE REVIEW

File: schema/go.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package schema
```
Comment (himank) [inline schema/go.go:1]: Copyright headers missing.

Comment (efirs) [reply schema/go.go:15]: Fixed. Few other lints are failing too. Working on fixing them.

**Comment by efirs** [comment]
Fixed as per feedback:

- [x] single .java file can have only one public non static class. I suggest making the main class public and rest of the model class non public (just remove public and let it fall back to default access modifier)
- [x] model classes needs to have equals and hashcode implemented to avoid user errors.
- [x] nice to have: model classes's can simplify user's life if they can generate default (no-arg) and one with all args constructor
- [x] the getters/setter - how do you generate? Java's standard for boolean field for accessors is to use isXxx for example for private boolean active; it will be public boolean isActive() and public void setActive(boolean active)
- [x] maven config - covers ~85% of the Java projects, some uses gradle,buck bazel etc... it will be best to link https://mvnrepository.com/artifact/com.tigrisdata/tigris-client and then user can figure out the snippet of their own dep manager
- [x] Nested objects should not inherit from TigrisCollectionType for TS and Java
- [x] Use JSON field names as language field names for TS ans Java

**Comment by ovaistariq** [comment]
TS/JS users are used to the npm/x toolchain. I wonder if this would be a friction point for them if we ask them to scaffold using a tool outside of their preferred toolchain.

**Review by himank — APPROVED** [review]

**Comment by himank** [comment]
> TS/JS users are used to the npm/x toolchain. I wonder if this would be a friction point for them if we ask them to scaffold using a tool outside of their preferred toolchain.

Do you mean to have `tigris scaffold` in npm? Something like `npm tigris scaffold...`

**Comment by efirs** [comment]
> > TS/JS users are used to the npm/x toolchain. I wonder if this would be a friction point for them if we ask them to scaffold using a tool outside of their preferred toolchain.
> 
> Do you mean to have `tigris scaffold` in npm? Something like `npm tigris scaffold...`

We can check what we can do to further improve it.

There are few possibilites:
* Invoke Tigris CLI by npx
* Reimplement in TypeScript
* Make this server side feature

**Pull request merged**